aspnetcore/blazor/call-web-api.md

```diff
@@ -0,0 +1,230 @@
+
+  In the following code, `_newItemName` is provided by a bound element of the component. The `AddItem` method is triggered by selecting a `<button>` element. See the sample app for a complete example.
+
+  ```cshtml
+  @using Microsoft.AspNetCore.Blazor.Http
+  @inject HttpClient Http
+
+  <input type="text" bind="@_newItemName" placeholder="New Todo Item" />
+  <button onclick="@(async () => await AddItem())">Add</button>
+
+  @functions {
+      private const string ServiceEndpoint = "https://localhost:10000/api/todo";
+      private string _newItemName;
+
+      private async Task AddItem()
+      {
+          var addItem = new TodoItem { Name = _newItemName, IsComplete = false };
+          await Http.PostJsonAsync(ServiceEndpoint, addItem);
+      }
+  }
+  ```
+
+* `PutJsonAsync` &ndash; Sends a PUT request, including JSON-encoded content.
+
+  In the following code, `_editItem` values (`Id`, `Name`, `IsCompleted`) are provided by bound elements of the component. The `SaveItem` method is triggered by selecting the Save `<button>` element. See the sample app for a complete example.
+
+  ```cshtml
+  @using Microsoft.AspNetCore.Blazor.Http
+  @inject HttpClient Http
+
+  <input type="text" bind="@_editItem.Id" />
+  <input type="checkbox" bind="@_editItem.IsComplete" />
+  <input type="text" bind="@_editItem.Name" />
+  <button onclick="@(async () => await SaveItem())">Save</button>
+
+  @functions {
+      private const string ServiceEndpoint = "https://localhost:10000/api/todo";
+      private TodoItem _editItem = new TodoItem();
+
+      private async Task SaveItem()
+      {
+          await Http.PutJsonAsync(
+              Path.Combine(ServiceEndpoint, _editItem.Id.ToString()), _editItem);
+      }
+  }
+  ```
+
+<xref:System.Net.Http> includes additional extension methods for sending HTTP requests and receiving HTTP responses. [HttpClient.DeleteAsync](xref:System.Net.Http.HttpClient.DeleteAsync*) is used to send a DELETE request to a web API.
+
+In the following code, the Delete `<button>` element supplies the `id` when it's selected in the UI. See the sample app for a complete example.
+
+```cshtml
+@using Microsoft.AspNetCore.Blazor.Http
+@inject HttpClient Http
+
+<input type="text" bind="@_id" />
+<button onclick="@(async () => await DeleteItem())">Delete</button>
+
+@functions {
+    private const string ServiceEndpoint = "https://localhost:10000/api/todo";
+    private long _id;
+
+    private async Task DeleteItem()
+    {
+        await Http.DeleteAsync(Path.Combine(ServiceEndpoint, _id.ToString()));
+    }
+}
+```
+
+## HttpClient and HttpRequestMessage with Fetch API request options
+
+Use [HttpClient](xref:fundamentals/http-requests) and <xref:System.Net.Http.HttpRequestMessage> to supply request options to the underlying JavaScript [Fetch API](https://developer.mozilla.org/docs/Web/API/Fetch_API).
```

Comment:
Most of this stuff is just the normal way you use `HttpClient`, but the `FetchArgs` stuff is special.

```suggestion
Use [HttpClient](xref:fundamentals/http-requests) and <xref:System.Net.Http.HttpRequestMessage> to customize requests. 

When running on WebAssembly, supply request options to the underlying JavaScript [Fetch API](https://developer.mozilla.org/docs/Web/API/Fetch_API) using the `WebAssemblyHttpMessageHandler.FetchArgs` property on the request. As shown in the following example, the `credentials` property is set to any of the following values:
```